src/main/kotlin/org/rust/cargo/project/configurable/RustfmtConfigurable.kt

```diff
@@ -5,13 +5,63 @@
 
 package org.rust.cargo.project.configurable
 
+import com.intellij.execution.configuration.EnvironmentVariablesComponent
+import com.intellij.openapi.options.BoundConfigurable
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.ComboBox
 import com.intellij.openapi.ui.DialogPanel
+import com.intellij.ui.RawCommandLineEditor
+import com.intellij.ui.components.Label
 import com.intellij.ui.layout.panel
+import com.intellij.ui.layout.toBinding
 import org.rust.RsBundle
+import org.rust.cargo.project.settings.rustfmtSettings
+import org.rust.cargo.toolchain.RustChannel
+
+class RustfmtConfigurable(project: Project) : BoundConfigurable(RsBundle.message("settings.rust.rustfmt.name")) {
+    private val state = project.rustfmtSettings.state
+
+    private val additionalArguments = RawCommandLineEditor()
+
+    private val channelLabel = Label(RsBundle.message("settings.rust.rustfmt.channel.label"))
+    private val channel = ComboBox<RustChannel>().apply {
+        RustChannel.values()
+            .sortedBy { it.index }
+            .forEach { addItem(it) }
+    }
+
+    private val environmentVariables = EnvironmentVariablesComponent()
 
-class RustfmtConfigurable(project: Project) : RsConfigurableBase(project, RsBundle.message("settings.rust.rustfmt.name")) {
     override fun createPanel(): DialogPanel = panel {
+        blockRow {
+            row(RsBundle.message("settings.rust.rustfmt.additional.arguments.label")) {
+                additionalArguments(pushX, growX)
+                    .comment(RsBundle.message("settings.rust.rustfmt.additional.arguments.comment"))
+                    .withBinding(
+                        componentGet = { it.text },
+                        componentSet = { component, value -> component.text = value },
+                        modelBinding = state::additionalArguments.toBinding()
```

Comment:
Done.

I'm curious since `state` in `SimplePersistentStateComponent` has the `@Volatile` annotation, shouldn't we mark all `RustfmtState` fields as `@Volatile`, since we are modifying them directly?

Reply:
> I'm curious, `state` in `SimplePersistentStateComponent` has the `@Volatile` annotation, shouldn't we mark all `RustfmtState` fields as `@Volatile`, since we are modifying them directly?

No, we shouldn't. If you use `BaseState` to create state property and modify the property, it triggers `com.intellij.openapi.components.BaseState#incrementModificationCount` (except some complex cases described in doc comment) that should be enough for synchronization